tests/masking_test.py

```diff
@@ -83,6 +145,91 @@ def cat(x, y, z):
         return lax.concatenate([x, y, x], 0)
     self.assertRaisesRegex(ShapeError, "", thunk)
 
+  def test_shapecheck_device_put(self):
+    @shapecheck(['n'], 'n')
+    def d_put(x):
+      return api.device_put(x)
+
+  def test_shapecheck_broadcast_in_dim(self):
+    x = np.zeros((7, 1))
+    lax.broadcast_in_dim(x, shape=(3, x.shape[0], 4), broadcast_dimensions=(1, 2))
+    @shapecheck(['(n, 1)'], '(3, n, 4)')
+    def broadcast_in_dim(x):
+      return lax.broadcast_in_dim(x, shape=(3, x.shape[0], 4), broadcast_dimensions=(1, 2))
+
+  def test_shapecheck_jit(self):
+    @shapecheck(['n'], '2*n')
+    @jit
+    def concat(x):
+      return lax.concatenate([x, x], 0)
+
+  def test_shapecheck_pad(self):
+    @shapecheck(['n'], '2*n+1')
+    def p(x):
+      return lax.pad(x, 0, [(1, 1, 1)])
```

Comment:
Like Chomsky's poverty of the stimulus, these checks only ensure that correct programs don't cause errors, but they don't check that incorrect programs do cause errors. It's not necessary for this PR, but we should probably add tests for the negative side too.